RUFAS/input_manager.py

```diff
@@ -19,19 +25,58 @@ def __init__(self) -> None:
         if InputManager.__instance is None:
             InputManager.__instance = self
         self.__metadata: Dict[str, Any] = {}
+        self.__pool: Dict[str, Any] = {}
 
     def _load_metadata(self, metadata_path: str = "input/example_metadata.json") -> None:
         """
         Loads metadata from json file to IM metadata object
 
         Parameters
         ----------
-            metadata_path : str
-                The path to the metadata file
+        metadata_path : str
+            The path to the metadata file.
+
+        Raises
+        ------
+        Exception
+            If an error occurs while opening or reading the metadata_path file.
+
         """
         try:
             with open(metadata_path) as metadata_file:
                 self.__metadata = json.load(metadata_file)
 
         except Exception as e:
             raise e
+
+    def _load_data(self) -> None:
+        """Loads data from JSON or CSV fileRaises
+
+        Raises
+        ------
+        Exception
+            If an error occurs while opening or reading a data file.
+
+        """
+        metadata_files_key = "files"
+        data_files = self.__metadata[metadata_files_key]
```

Comment:
```suggestion
        files_details = self.__metadata["files"]
```
I think this is an overkill.
Also changed the name to a more descriptive one.